src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -102,6 +102,20 @@
     /** Maximum time we allow for elapsed fsync before WARNing */
     private final static long fsyncWarningThresholdMS;
 
+    /**
+     * This parameter limit the size of each txnlog to a given limit (KB).
+     * It does not affect how often the system will take a snapshot
+     * [zookeeper.snapCount]
+     *
+     * The feature is disabled by default (-1)
+     */
+    public static final String LOG_SIZE_LIMIT = "zookeeper.txnlogSizeLimit";
+
+    /**
+     * The actual txnlog size limit in bytes.
+     */
+    public static long logSizeLimit = -1;
+
```

Comment:
private? txnLogSizeLimit is better?

Reply:
good points @maoling ! is there a reason these two variables should be public? it would be nice to make the variable have a similar name to the property.